src/index.js

```diff
@@ -62,6 +65,68 @@ const createContext = (ctx) => {
   return ctx
 }
 
+const addTypeScriptLoader = (options = {}, loader) => {
+  const moduleName = 'postcss'
+
+  return {
+    ...options,
+    searchPlaces: [
+      ...(options.searchPlaces || []),
+      'package.json',
+      `.${moduleName}rc`,
+      `.${moduleName}rc.json`,
+      `.${moduleName}rc.yaml`,
+      `.${moduleName}rc.yml`,
+      `.${moduleName}rc.ts`,
+      `.${moduleName}rc.js`,
+      `${moduleName}.config.ts`,
+      `${moduleName}.config.js`
+    ],
+    loaders: {
+      ...options.loaders,
+      '.ts': loader
+    }
+  }
+}
+
+const withTypeScriptLoader = (rcFunc) => {
+  let registerer
+
+  // Register TypeScript compiler instance
+  try {
+    registerer = require('ts-node').register()
```

Comment:
Can we call `require('ts-node')` only if `.ts` file as found?

I do not want to load heavy TypeScript for the projects with `.js` configs, but with `ts-node` in the system.

Reply:
I'll have a look, but this doesn't actually load the heavy side of the TypeScript compiler until it's actually needed, making it a very cheap call.

Reply:
Am I right that we are calling `require('ts-node').register()` on any config loading?

(I may be wrong, I am not sure)

Reply:
Yes; like I said the require cost is negligible since it doesn't actually do any work (including creating the TypeScript compiler, and compiling code, which are the expensive operations) until a `.ts` file is required.

However I think we *might* be able to move it into the `loader` function.